server.cpp

```diff
@@ -38,6 +38,19 @@ std::string make_daytime_string()
     return buffer;
 }
 
+std::string int_to_hex(int dec)
+{
+    std::string digits[16] = {"0","1","2","3","4","5","6","7","8","9","A","B","C","D","E","F"};
+    std::string hex;
+    do
+    {
+        hex.insert (0, digits[dec % 16]);
+        dec /= 16;
+    }
+    while (dec != 0);
```

Comment:
`do ... while` cycles are considered unnatural and potentially harmful. Consider rewriting it to `for` cycle.